src/main/java/org/jboss/tools/intellij/analytics/CliReleaseDownloader.java

```diff
@@ -0,0 +1,120 @@
+package org.jboss.tools.intellij.analytics;
+
+import com.intellij.ide.plugins.IdeaPluginDescriptor;
+import com.intellij.ide.plugins.PluginManager;
+import com.intellij.openapi.extensions.PluginId;
+import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.util.io.HttpRequests;
+import com.redhat.devtools.intellij.telemetry.core.service.TelemetryMessageBuilder.ActionMessage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.BufferedInputStream;
+
+import java.util.zip.GZIPInputStream;
+
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+import org.apache.commons.compress.archivers.tar.TarArchiveInputStream;
+import org.apache.commons.compress.utils.IOUtils;
+
+
+public final class CliReleaseDownloader {
```

Comment:
Instead of a new class to download Cli, can't be the https://github.com/redhat-developer/intellij-dependency-analytics/blob/main/src/main/java/org/jboss/tools/intellij/analytics/GitHubReleaseDownloader.java reused after injecting the repo to the construtor?

Of course the untar part has to be done externally if you reuse GitHubReleaseDownloader.

Other option is to inherit a untare downloader from GitHubReleaseDownloader.

e.g. 
```
class GitHubTarReleaseDownloader extends GitHubReleaseDownloader {
...
@Override
public File download(final ProgressIndicator indicator) throws IOException {
  final File tarball = super.download(indicator);
  untar(tarball);
  ...
  return cliPath
}
}
```